tick/solver/saga.py

```diff
@@ -205,6 +208,24 @@ def set_model(self, model: ModelGeneralizedLinear):
             The `Solver` with given model
         """
         if isinstance(model, ModelGeneralizedLinear):
+            # Construct the wrapped C++ SAGA solver
+
+            first = self.dtype is None or self.dtype != model.dtype
+
+            self.dtype = model.dtype
+
+            step = self.step
+            if step is None:
+                step = 0.
+            epoch_size = self.epoch_size
+            if epoch_size is None:
+                epoch_size = 0
+
+            self._solver = dtype_class_mapper[self.dtype](
+                epoch_size, self.tol, self._rand_type, step, self.seed)
+            if first is True:
```

Comment:
This should be done even if first is False.

Reply:
I don't think so, if the dtype is the same there's no need to reset variance_reduction right?